src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -41,5 +66,88 @@ private void PreferencesPanel_MouseDown(object sender, MouseButtonEventArgs e)
+            var saveChangesButton = sender as Button;
+
+            //Get the Grid that contains the Stack Panel that also contains the info related to the new style
+            var grid = (saveChangesButton.Parent as Grid).Parent as Grid;
+
+            var groupNameLabel = grid.FindName("groupNameBox") as TextBox;
+
+            var colorHexString = grid.FindName("colorHexVal") as Label;
+
+            var newItem = new StyleItem() { GroupName = groupNameLabel.Text, HexColorString = colorHexString.Content.ToString() };
+
+            if (string.IsNullOrEmpty(newItem.GroupName))
+                newItem.GroupName = "Input";
+
+            //if the validation returns false it means that the new style that will be added doesn't exists
+            if (viewModel.ValidateExistingStyle(newItem) == false)
+            {
+                viewModel.StyleItemsList.Add(newItem);
+                viewModel.ResetAddStyleControl();
+                AddStyleBorder.Visibility = Visibility.Collapsed;
+                AddStyleButton.IsEnabled = true;
+            }
+            else
+            {
+                viewModel.IsWarningEnabled = true;
+            }
+            
+        }
+
+        private void AddStyle_CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            AddStyleBorder.Visibility = Visibility.Collapsed;
+            AddStyleButton.IsEnabled = true;
+            viewModel.ResetAddStyleControl();
+        }
+
+        private void removeStyle_Click(object sender, RoutedEventArgs e)
+        {
+           var removeButton = sender as Button;
+
+            //Get the Grid that contains all the buttons in the StyleItem
+           var grid = (removeButton.Parent as Grid).Parent as Grid;
+
+            //Find inside the Grid the label that contains the GroupName (unique id)
+           var groupNameLabel = grid.FindName("groupNameLabel") as Label;
+
+            //Remove the selected style from the list
+            viewModel.RemoveStyleEntry(groupNameLabel.Content.ToString());
+        }
+
+        private void buttonColorPicker_Click(object sender, RoutedEventArgs e)
+        {
+            System.Windows.Forms.ColorDialog colorDialog = new System.Windows.Forms.ColorDialog();
+
+            if (colorDialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                Button colorButton = sender as Button;
+                if (colorButton != null)
+                    colorButton.Background = new SolidColorBrush(Color.FromRgb(colorDialog.Color.R, colorDialog.Color.G, colorDialog.Color.B));
+            }
+        }
+
+        private void Expander_Expanded(object sender, RoutedEventArgs e)
+        {
+            Expander currentExpander = sender as Expander;
+            Grid parentGrid = currentExpander.Parent as Grid;
+            foreach (Expander expander in parentGrid.Children)
+            {
+                if (expander != currentExpander)
+                    expander.IsExpanded = false;
```

Comment:
Thanks! Although this will be covered in a different task, no harm to include and this is what I imagined how we would implement, thanks!

Reply:
@Astul-Betizagasti see that I implemented the expanders functionality in case you need to use it in other places.
Thanks